src/providers/ConfigCodeActionProvider.ts

```diff
@@ -0,0 +1,223 @@
+      );
+
+      if (checkGateMatch) {
+        for (const match of checkGateMatch) {
+          seen[match] ? ++seen[match] : (seen[match] = 1);
+          const matchIndex =
+            nthIndexOf(searchableText, match, seen[match]) + offset;
+          const matchRange = new vsc.Range(
+            doc.positionAt(matchIndex),
+            doc.positionAt(matchIndex + match.length),
+          );
+          edit.replace(doc.uri, matchRange, checkGateReplacement(language));
+          changesMade = true;
+        }
+      }
+
+      const getConfigMatch = searchableText.match(
+        getStatsigAPIGetConfigRegex(language, config),
+      );
+
+      if (getConfigMatch) {
+        for (const match of getConfigMatch) {
+          seen[match] ? ++seen[match] : (seen[match] = 1);
+          const matchIndex =
+            nthIndexOf(searchableText, match, seen[match]) + offset;
+          const matchRange = new vsc.Range(
+            doc.positionAt(matchIndex),
+            doc.positionAt(matchIndex + match.length),
+          );
+          edit.replace(doc.uri, matchRange, getConfigReplacement(language));
+          changesMade = true;
+        }
+      }
+      const variableAssignmentMatch = searchableText.match(
+        getVariableAssignmentRegex(language, config),
+      );
+      if (variableAssignmentMatch) {
+        for (const match of variableAssignmentMatch) {
+          seen[match] ? ++seen[match] : (seen[match] = 1);
+          const matchIndex =
+            nthIndexOf(searchableText, match, seen[match]) + offset;
+          const line = doc.lineAt(doc.positionAt(matchIndex).line);
+          edit.delete(doc.uri, line.rangeIncludingLineBreak);
+          changesMade = true;
+        }
+      }
+    });
+
+    return changesMade;
+  }
+}
+
+const removeStaleConfigCommandHandler = (config: string, success: boolean) => {
+  if (success) {
+    void vsc.window.showInformationMessage(`Removed ${config}`);
+  } else {
+    void vsc.window.showInformationMessage(`Could not remove ${config}`);
+  }
+};
+
+const removeAllStaleConfgsCommandHandler = async () => {
+  const output = vsc.window.createOutputChannel('Statsig output');
+  const edit = new vsc.WorkspaceEdit();
+  const files = await vsc.workspace.findFiles(
+    `**/*.{${SUPPORTED_FILE_EXTENSIONS.join()}}`,
+  );
+  const cleanedFiles = [];
+  const configCounts: { [key: string]: number } = {};
+  for (const file of files) {
+    const doc = await vsc.workspace.openTextDocument(file);
+    const staleConfigs = findStaleConfigs(doc);
+    if (staleConfigs.length === 0) {
+      continue;
+    }
+    if (ConfigCodeActionProvider.handleEdit(doc, edit, staleConfigs)) {
+      staleConfigs.forEach((config) => {
+        configCounts[config]
+          ? ++configCounts[config]
+          : (configCounts[config] = 1);
+      });
+      cleanedFiles.push(file.path);
+      output.appendLine(`(success) ${file.path}`);
+    } else {
+      output.appendLine(`(failed) ${file.path}`);
```

Comment:
This case is specifically if we detect usage of a stale gate but cannot actually remove it (demonstrated in test plan). This happens when we don't have coverage of a certain syntax where the config can be used.